Rust/src/lib.rs

```diff
@@ -49,63 +61,69 @@ fn base_speed() -> f32 {
 mod tests {
     use super::*;
 
-    #[test]
-    fn european_parrot_speed() {
-        let parrot = Parrot { parrot_type: "european_parrot",
-                              number_of_coconuts: 0,
-                              voltage: 0.0,
-                              nailed: false };
-        assert_eq!(parrot.speed().unwrap(), 12.0);
+    pub struct TestData<'a> {
+        pub name: String,
+            input: Parrot<'a>,
+        pub expected: f32,
     }
 
-    #[test]
-    fn african_parrot_speed_with_one_coconut() {
-        let parrot = Parrot { parrot_type: "african_parrot",
-                              number_of_coconuts: 1,
-                              voltage: 0.0,
-                              nailed: false };
-        assert_eq!(parrot.speed().unwrap(), 3.0);
+    impl TestData<'_> {
+        pub fn new(name: impl Into<String>, input: Parrot, expected: f32) -> TestData {
```

Comment:
> Did you use this trait so that you could pass static string to the new function? I looked at the code and it seemed like the type should be &str but then i ran into all sorts of lifetime stuff - I guess you found this as well?

I did this in my previous Rust submission as well, as I found that like Go you can't create an inline/anonymous String and pass a reference to it in one line, and I didn't need a pointer, so I didn't really want to do that. After some quick Googling I came across this suggestion, and have used it since, no idea if that's considered bad or non-idiomatic in Rust :man_shrugging: